src/sort/index.ts

```diff
@@ -1,33 +1,57 @@
-import { alias, configurable, tag, Events, Selectors, Store, Tag } from '@storefront/core';
+import { alias, configurable, tag, Events, Selectors, Store, StoreSections, Tag } from '@storefront/core';
 
 @configurable
 @alias('sort')
 @tag('gb-sort', require('./index.html'))
 class Sort {
 
   props: Sort.Props = {
-    labels: []
+    labels: [],
   };
   state: Sort.State = {
     sorts: [],
-    // TODO make this better by fixin the sort action
-    onSelect: (index) => this.actions.selectSort(index)
+    onSelect: (index) => {
+      switch (this.props.storeSection) {
+        case StoreSections.PAST_PURCHASES:
+          this.actions.selectPastPurchasesSort(index);
+          break;
+        case StoreSections.SEARCH:
+          this.actions.selectSort(index);
+      }
+    }
   };
 
   init() {
     this.updateSorts();
-    this.flux.on(Events.SORTS_UPDATED, this.updateSorts);
+    switch (this.props.storeSection) {
```

Comment:
can we abstract this switch logic in the short term if we have to keep reusing it in all the components?

Reply:
I don't think the switch logic could easily be abstracted out without refactoring the selectors/events entirely as only the component can know what store sections relate to what actions need to be listened to.